Simple-Projects/Simple_Projects/Basic/List/List.h

```diff
@@ -1,8 +1,7 @@
 #ifndef LIST_H
 #define LIST_H
 
-#include <iostream>
-#include <cstdlib>
+#include <iosfwd>
 
 class OutOfMemoryException {};
```

Comment:
We need to talk about exceptions. As it is now, your exceptions are a bit unwieldy :)
Most importantly, if you introduce custom exceptions, you probably want them to inherit (publicly) from std::exception and then be able to catch anything that might go wrong as catch(std::exception& exc) - the objects will be upcast to the base reference